src/dune_rules/jsoo_rules.ml

```diff
@@ -72,6 +72,53 @@ end = struct
       | name, false -> [ "--disable"; name ])
 end
 
+module Version = struct
+  type t = int list
+
+  let of_string s : t =
+    let s =
+      match
+        String.findi s ~f:(function
+          | '+' | '-' | '~' -> true
+          | _ -> false)
+      with
+      | None -> s
+      | Some i -> String.take s i
+    in
+    String.split s ~on:'.' |> List.map ~f:int_of_string
```

Comment:
Can this function just return an option directly?

Reply:
changed